src/pages/Stats.jsx

```diff
@@ -0,0 +1,260 @@
+import React from 'react'
+import AlertContainer from 'react-alert'
+import moment from 'moment'
+import HeaderBar from '../components/HeaderBar'
+import TitleSection from '../components/TitleSection'
+import strings from '../constants/strings'
+import * as colors from '../constants/colors'
+import Loading from '../components/Loading'
+import 'react-select/dist/react-select.css'
+import './Stats.scss'
+
+
+class Stats extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      schedules: null,
+      search: ''
+    }
+
+    this.weeklySchedule = this.weeklySchedule.bind(this);
+  };
+
+  componentWillMount() {
+    let confirmedURL = 'https://cit-stats.herokuapp.com/services/confirmed';
+    let confirmedPromise = fetch(confirmedURL)
+      .then(res => res.json())
+      .then(res => {
+
+        return res
+          .map((val, i) => ({
+            people: val.people,
+            type: val.type,
+            createdAt: val.createdAt,
+            updatedAt: val.updatedAt
+          }));
+      })
+
+    let declinedURL = 'https://cit-stats.herokuapp.com/services/declined';
+    let declinedPromise = fetch(declinedURL)
+      .then(res => res.json())
+      .then(res => {
+
+        return res
+          .map((val, i) => ({
+            people: val.people,
+            type: val.type,
+            createdAt: val.createdAt,
+            updatedAt: val.updatedAt
+          }));
+
+      })
+
+    Promise.all([confirmedPromise, declinedPromise]).then(values => {
+      //console.log(values);
+      let confirmed = values[0];
+      let declined = values[1];
+      let volunteers = this.weeklySchedule(values);
+      this.badEggs(volunteers);
+
+    });
+
+  }
+
+  weeklySchedule(values) {
+    let weekly = []
+    let confirmed = values[0]
+    let declined = values[1]
+    // returning daily data to weekly
+    for (var i = 0; i<confirmed.length; i=i+7) {
+      weekly.push(confirmed[i])
+    }
+
+    for (var i = 0; i<declined.length; i=i+7) {
+        weekly.push(declined[i])
+    }
+
+    // returning schedule by volunteer ID
+    var data = []
+    let volunteers = {}
+
+    for (var ind = 0; ind<weekly.length; ind++) {
+      data = weekly[ind]
+
+      for (var index = 0, l = data['people'].length; index < l; index++) {
+        var obj = data['people'][index];
+
+          if (!(obj['_id'] in volunteers)) {
+
+            volunteers[obj['_id']] = {
+              'name': obj['name'],
+              'schedule': [{
+                 'updatedAt': data['updatedAt'],
+                 'type': data['type']
+              }]
+            }
+          }
+        volunteers[obj['_id']]['schedule'].push({
+          'updatedAt': data['updatedAt'],
+          'type': data['type']
+        })
+      }
+    }
+
+    // sorting schedule by updatedAt timestamp
+    for (var key of Object.keys(volunteers)) {
+      volunteers[key]['schedule'].sort(function(a,b){
```

Comment:
you can use an arrow function for this sort: `sort((a,b) => { your function })`. it should be a bit cleaner